### PR TITLE
Improve the DX by adding user-data normalisation and listing operations

### PR DESCRIPTION
**Pull request by RDPerera**
This PR introduces two improvements:  

1. **Data Normalization for `UploadUserData` Operation:**  
   - Added support for uploading user data as a JSON array.  
   - Implemented data normalization to handle various formats of user identifiers (e.g., emails, phone numbers, mobile IDs, third-party user IDs, and address information).  
   - JSON arrays are internally converted to [Google Ads API `UserData`](https://developers.google.com/google-ads/api/rest/reference/rest/v17/UserData) objects.  

2. **New Getter Operations with Filtering:**  
   - Implemented `getCustomers` to retrieve customer data.  
   - Implemented `getCampaigns` to retrieve campaign data.  
   - Implemented `getUserLists` with a **filterByName** option for filtering lists by name.  

#### Details of the UploadUserData Improvement  

The `UploadUserData` operation now supports uploading user data as a JSON array with the following acceptance criteria:  

- **Input Format:**  
  The input must be a JSON array of user objects. Each object should contain a single user identifier in one of the following formats:  
  ```json
  [
    {"user_identifier1": "value1"},
    {"user_identifier2": "value2"},
    {"user_identifier3": "value3"}
  ]
  ```  

- **Internal Conversion:**  
  JSON arrays are normalized and converted to the `UserData` object schema defined by the [[Google Ads API](https://developers.google.com/google-ads/api/rest/reference/rest/v17/UserData)](https://developers.google.com/google-ads/api/rest/reference/rest/v17/UserData).  
  - Each user object in the JSON array is mapped to `UserIdentifier` fields in the `UserData` object.  
  - Fields such as `hashedEmail`, `hashedPhoneNumber`, `addressInfo`, etc., are seamlessly integrated.  

- **Supported User Identifiers:**  
  - **Phone Number:**  
    Examples:  
    ```json
    {"phoneNumber": "+94112132234"}  
    {"hashedPhoneNumber": "hashed_value"}  
    ```  

  - **Email Address:**  
    Examples:  
    ```json
    {"email": "someone@sample.com"}  
    {"hashedEmail": "hashed_value"}  
    ```  

  - **Mobile ID:**  
    Example:  
    ```json
    {"mobileId": "23131234324"}  
    ```  

  - **Third-Party User ID:**  
    Example:  
    ```json
    {"thirdPartyUserId": "2124234234"}  
    ```  

  - **Address Information:**  
    Examples:  
    ```json
    {"addressInfo": {"firstName": "Someone", "city": "Colombo", "postalCode": "12345"}}  
    {"hashedAddressInfo": {"hashedFirstName": "hashed_value", "city": "Colombo"}}  
    ```  

#### Getter Operations with FilterByName  

- **`getUserLists`:**  
  - Fetches user lists with an optional `filterByName` parameter.  
  - If `filterByName` is provided, only lists matching the given name pattern are returned.  

- **`getCustomers`:** Retrieves customer data with relevant details.  
- **`getCampaigns`:** Retrieves campaign details with filtering options.  

#### Examples  
**UploadUserData:**  
Input:  
```json
[
  {"email": "user1@domain.com"},
  {"phoneNumber": "+94112132234"},
  {"addressInfo": {"firstName": "John", "city": "Colombo", "postalCode": "12345"}}
]
```  
Internal Conversion to `UserData`:  
```json
[
  {
    "userIdentifiers": [{"emailAddress": "user1@domain.com"}]
  },
  {
    "userIdentifiers": [{"phoneNumber": "+94112132234"}]
  },
  {
    "userIdentifiers": [{
      "addressInfo": {"firstName": "John", "city": "Colombo", "postalCode": "12345"}
    }]
  }
]
```

### CODE REVIEW

File: .connector-store/meta.json
Comment (arunans23) [inline .connector-store/meta.json:1]: Please check whether the changes in this file are needed.

File: pom.xml [hunk 1]
```diff
@@ -23,9 +23,9 @@
          xmlns="http://maven.apache.org/POM/4.0.0"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
 
-    <groupId>org.wso2.integration.connector</groupId>
-    <artifactId>mi-connector-googleads</artifactId>
-    <version>1.0.3-SNAPSHOT</version>
+    <groupId>org.wso2.carbon.esb.connector</groupId>
```
Comment (arunans23) [inline pom.xml:26]: Lets keep the groupId, artifactId as it is.

File: src/main/java/org/wso2/carbon/google/ads/connector/JSONContentProcessor.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        ObjectMapper mapper = new ObjectMapper();
+        ArrayNode payload = mapper.createArrayNode();
+        ObjectNode userData = mapper.createObjectNode();
+        userData.putArray("userIdentifiers");
+
+        try {
+            JsonNode jsonArray = mapper.readTree(jsonArrayContent);
+
+            if (!jsonArray.isArray()) {
+                throw new RuntimeException("Input content is not a valid JSON array");
```
[the remark this answers is not in the document]
Comment (RDPerera) [reply src/main/java/org/wso2/carbon/google/ads/connector/JSONContentProcessor.java]: Sure. Changed with [916a2ca](https://github.com/wso2-extensions/esb-connector-googleads/pull/8/commits/916a2ca8a9f3d476371dd8025f4e1cf5f6310b46).

File: pom.xml (hunk 1 above)
Comment (RDPerera) [reply pom.xml]: Changed with https://github.com/wso2-extensions/esb-connector-googleads/pull/8/commits/b438836341d2f4b990fe90eda3ab36f906c9d957. and updated according to the https://github.com/wso2-extensions/esb-connector-googleads/commit/86485d1555575e6ecb6a990105393a1fcda4d869.

File: .connector-store/meta.json
Comment (RDPerera) [reply .connector-store/meta.json:1]: Yes sure. Changed with https://github.com/wso2-extensions/esb-connector-googleads/pull/8/commits/b438836341d2f4b990fe90eda3ab36f906c9d957. and updated according to the https://github.com/wso2-extensions/esb-connector-googleads/commit/86485d1555575e6ecb6a990105393a1fcda4d869.

**Review by arunans23 — APPROVED** [review]

**Pull request merged**